src/wallet/walletdb.cpp

```diff
@@ -560,6 +562,17 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             libzcash::SproutSpendingKey key;
             ssValue >> key;
 
+            // Previous versions of zcash did not checksum stored keys. Try to deserialize a checksum hash and verify
+            // it if deserialization is successful.
+            try {
+                uint256 hashChecksum;
+                if (SproutKeyChecksum(addr, key) != hashChecksum) {
```

Comment:
Pretty sure the refactor from 5df1c64 makes it so this will always evaluate false. It compares an empty `uint256` to the checksum. It needs to try to deserialize the checksum value before doing the comparison (as is done below with the sapling key checksum deserialization).

Reply:
Yes this introduced a bug. Thanks for catching it.

Reply:
Resolved with 51622c067